src/games/strategy/triplea/ai/proAI/ProAI.java

```diff
@@ -338,7 +346,7 @@ public Territory retreatQuery(final GUID battleID, final boolean submerge, final
         && (battleTerritory.isWater() || Match.someMatch(attackers, Matches.UnitIsLand))) {
       return null;
     }
-    ProBattleUtils.setData(getGameData());
+    calc.setData(getGameData());
```

Comment:
So essentially at the start of each phase you need to copy the current game data into the calc. This should only be done once since its slow. In this case, retreatAI.retreatQuery is when the calc will be used to check predicted battle outcome and determine whether to retreat.